crypto/src/lib.rs

```diff
@@ -44,17 +43,26 @@ pub mod bls12377;
 #[path = "tests/bls12381_tests.rs"]
 pub mod bls12381_tests;
 
+// Signing traits
+pub mod traits;
+// Key scheme implementations
 pub mod bls12381;
 pub mod ed25519;
+pub mod secp256k1;
+// Other tooling
 pub mod hkdf;
 pub mod pubkey_bytes;
 pub mod serde_helpers;
 
-pub mod secp256k1;
-
-pub mod traits;
-
-pub type CryptoError = ed25519_dalek::ed25519::Error;
+// Here we select the types that are used by default in the code base.
+// The whole code base should only:
+// - refer to those aliases and not use the individual scheme implementations
+// - not use the schemes in a way that break genericity (e.g. using their Struct impl functions)
+// - swap one of those aliases to point to another type if necessary
+pub type PublicKey = ed25519::Ed25519PublicKey;
```

Comment:
Does this mean that projects which use a different scheme will have to fork our crate and update these types with their preferred algorithm?

Reply:
That depends. Projects that want to use our crypto library won't have to (in fact it should soon live under mysten-infra). Projects that want to use Narwhal with a different scheme will have to, yes.
We can play a conditional compilation game here (reminiscent of Dalek backend selection) but it has severe limitations (as you know features are additive).